server/src/main/scala/org/http4s/server/middleware/RequestLogger.scala

```diff
@@ -0,0 +1,29 @@
+package org.http4s
+package server
+package middleware
+
+import fs2._
+import org.log4s._
+import cats.implicits._
+import fs2.interop.cats._
+
+/**
+  * Simple Middleware for Logging Requests As They Are Processed
+  */
+object RequestLogger {
+  private[this] val logger = getLogger
+
+  def apply(service: HttpService)(implicit strategy: Strategy): HttpService = Service.lift{ req =>
+    Stream(req)
+      .observe(requestLogSink)
+      .evalMap[Task, Task, MaybeResponse]{req => service(req)}
+      .runFoldMap(identity)
```

Comment:
I tried that approach, although through is `Pipe` and Blind Alley of Death is `Sink` and I never got a rendered body. It would display the correct headers, but the body didn't come through. Again this was a couple hours work so it's probably doable. This works, surprisingly. Every version where I attempted a copy I ended up failing to render the body.